src/main/java/com/maxmind/minfraud/request/Email.java

```diff
@@ -14,6 +20,29 @@
     private final String address;
     private final boolean hashAddress;
     private final String domain;
+    private static final Map<String, String> typoDomains;
+    private static final Pattern addressDashRegex;
+    private static final Pattern addressPlusRegex;
+
+    static {
+        HashMap<String, String> m = new HashMap<>();
+
+        // gmail.com
+        m.put("35gmai.com", "gmail.com");
+        m.put("636gmail.com", "gmail.com");
+        m.put("gamil.com", "gmail.com");
+        m.put("gmail.comu", "gmail.com");
+        m.put("gmial.com", "gmail.com");
+        m.put("gmil.com", "gmail.com");
+        m.put("yahoogmail.com", "gmail.com");
+        // outlook.com
+        m.put("putlook.com", "outlook.com");
+
+        typoDomains = Collections.unmodifiableMap(m);
+
+        addressDashRegex = Pattern.compile("\\A([^-]+)-.*\\z");
+        addressPlusRegex = Pattern.compile("\\A([^+]+)\\+.*\\z");
```

Comment:
Isn't a regex a bit of overkill for this, especially in languages where they are heavy both in terms of syntax and performance? In the [Python PR](https://github.com/maxmind/minfraud-api-python/pull/74/files#diff-2957d6cf018deb6b6f6cd0105db155e6d77d74358cb92144cdb34f6db6aa099aR101-R108), I just found the location of `-`/`+` and then took a substring.

Reply:
Yeah, that's true. I'll update it. Thanks!